src/compiler/checker.ts

```diff
@@ -6822,6 +6985,8 @@ module ts {
          * If 'node' is a CallExpression or a NewExpression, then its argument list is returned.
          * If 'node' is a TaggedTemplateExpression, a new argument list is constructed from the substitution
          *    expressions, where the first element of the list is the template for error reporting purposes.
+         * If 'node' is a Decorator, a new argument list is constructed with the decorator
+         *    expression as a placeholder.

```

Comment:
Are these also used for error reporting purposes? If so, maybe throw that in there.